### PR TITLE
Swift compiler performance tweaks

### PR DESCRIPTION
**Pull request by koke**
Context: [Diving into Swift compiler performance](https://koke.me/2017/03/24/diving-into-swift-compiler-performance/)

Basically tweaking a few things to make type checking easier on the compiler. I removed the `NSIndexPath` extension rather than optimizing, because I don't think we needed that (cc @jleandroperez)

screen shot 2017-03-24 at 15 33 00

Needs review: @astralbodies

### CODE REVIEW

File: WordPress/Classes/Utility/WPTableViewHandler.m [hunk 1]
```diff
@@ -205,12 +205,12 @@ - (void)refreshTableViewPreservingOffset
 
 - (void)cacheRowHeight:(CGFloat)height forIndexPath:(NSIndexPath *)indexPath
 {
-    [self.cachedRowHeights setObject:@(height) forKey:[indexPath toString]];
+    [self.cachedRowHeights setObject:@(height) forKey:indexPath];
```
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPTableViewHandler.m]: I think this could be a problem. I remember debugging this, and it was the reason why `toString` was added as an extension: TableView may provide different instances of `NSIndexPath`, for the same coordinates, when you're scrolling.

And since indexPath isn't "Hashable", the cache might not be picked up. Can we revert this change?

File: WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift [hunk 2]
```diff
@@ -2,5 +2,5 @@ import Foundation
 
 class SigninEditingState {
     static var signinEditingStateActive = false
-    static var signinLastKeyboardHeightDelta = CGFloat(0.0)
+    static var signinLastKeyboardHeightDelta: CGFloat = 0
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: Does this actually optimize build time?

File: WordPress/Classes/Utility/WPTableViewHandler.m (hunk 1 above)
Comment (koke) [inline WordPress/Classes/Utility/WPTableViewHandler.m]: `IndexPath` is both `Hashable` and `Equatable`. As long as you're not comparing them with `==`, it should be fine.

File: WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift (hunk 2 above)
Comment (koke) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: Suprisingly yes. I guess `CGFloat` has 15 different constructors, and the compiler has to figure out which one it's the right one, while in the second case, there's an explicit type and a literal.

Comment (catehstn) [inline WordPress/Classes/ViewRelated/NUX/SigninEditingState.swift]: 15 different constructors 🙈😱

File: WordPress/Classes/Utility/WPTableViewHandler.m (hunk 1 above)
Comment (jleandroperez) [inline WordPress/Classes/Utility/WPTableViewHandler.m]: @koke This extension was introduced in [this commit](https://github.com/wordpress-mobile/WordPress-iOS/commit/bf2fcdd3a35fcfed3a1b32433b9134f435859905) ("`Notifications: Fixes a row height caching bug`").

Can't remember exactly *what* delegate method was producing this, but bottom line was: UIKit was passing on an internal NSIndexPath subclass, and the equality operator was failing on us. 

I've tried triggering the original bug, but odds are, it was either fixed, or... i can't remember the exact conditions.

Just verified, and cannot trigger the initial conditions. Good to go on my end!

P.s.: Hashable / Equatable are for the Swift struct realm!
